### ISSUE TITLE
Is bool a subtype of int?

### ISSUE THREAD

**Issue by LuKuangChen** [facebookincubator/cinder#46]
### What version of Static Python are you using?

4f1b313
2021-07-27

### What program did you run?

```python
x: int = True
```

### What happened?

The program errors.

```
compiler.static.errors.TypedSyntaxError: type mismatch: Exact[bool] cannot be assigned to int
```

### What should have happened?

We expected no error because `bool` is a subclass of `int` (`isinstance(True, int)` is `True`).

**Comment by DinoV** [comment]
We should be allowing this

**Comment by LuKuangChen** [comment]
In general, if C is a subclass of D, is C also a subtype of D?

**Comment by tekknolagi** [comment]
This should be fixed after 88dd90d1fd43b3c751c3557ba5fc3a140717c3b1. Thanks!
